### ISSUE TITLE
Installation setup steps

### ISSUE THREAD

**Issue by adelin-b** [systemd/zram-generator#99]
Hello, I think it would be nice to have installation steps described as once installed there is not indication of what to do beside creating a conf file.
I installed it and now I do not find if:
- User should enable `zram` module or is it done automatically
- How to enable the `systemd-zram-generator-setup@` after installing

**Comment by polarathene** [comment]
[This information might help you](https://github.com/systemd/zram-generator/issues/78), if it does or you have any trouble with it, provide feedback here.

---

For installation on an Ubuntu VM, I have been using the following:

```sh
apt-get -qq update && apt-get -qq install build-essential pkg-config ronn && \
curl https://sh.rustup.rs -sSf | sh -s -- -y &&\
source $HOME/.cargo/env && \
git clone https://github.com/systemd/zram-generator && \
cd zram-generator && make build && make install && \
echo -e "[zram0]\ncompression-algorithm = zstd\nzram-fraction = 2.0" > /etc/systemd/zram-generator.conf && \
systemctl daemon-reload
```

That installs dependencies and builds the current git `main` branch commit. Then the last line with `echo` creates a simple config for zram to use `zstd` with 2x RAM (default max is 4GB though).

If you install from a distro package instead, then a default config may be available and used, a list of locations and their priority to override other configs is [shown on the README](https://github.com/systemd/zram-generator#configuration), and an example config with detailed comments to understand it [is provided in the git repo files](https://github.com/systemd/zram-generator/blob/main/zram-generator.conf.example).

---

## TL;DR

- Install `zram-generator`.
- Create a `zram-generator.conf` config file.
- `systemctl daemon-reload` should be sufficient to create the zram device.
- `zramctl` will confirm.

Making changes to config file and applying that or disabling `zram-generator` is bit more effort IIRC.

If no package from distro, uninstalling is also bit more work.

---

> User should enable `zram` module or is it done automatically

This shouldn't be necessary. It should be handled automatically for you. Previously it was an issue and required `modprobe zram`.

> How to enable the `systemd-zram-generator-setup@` after installing

`systemctl daemon-reload`, if you need more than that, look at the linked issue I provided.

**Comment by 5taras** [comment]
### I tried to compile as polarathene wrote and get error: 
```
**[install -Dpm755 target/release/zram-generator -t /usr/lib/systemd/system-generators/
install: cannot create regular file '/usr/lib/systemd/system-generators/zram-generator': Permission denied
make: *** [Makefile:62: install] Error 1].**
```
If I run sudo make install instead I get:

```
sudo make install
cargo build --release 
make: cargo: No such file or directory
make: *** [Makefile:32: program] Error 127
```

**Comment by nabijaczleweli** [comment]
install cargo bestie

**Comment by 5taras** [comment]
Installed cargo. New error:

sudo make install
cargo build --release 
error: failed to parse lock file at: /home/therion/zram-generator/Cargo.lock

Caused by:
  lock file version 4 requires `-Znext-lockfile-bump`
make: *** [Makefile:32: program] Error 101
